### PR TITLE
Feat/private posts

### PR DESCRIPTION
**Pull request by asulis**
### Description:

This pull request implements Private Posts functionality (Issue #96 )

### Checklist:

_Make sure you tick all the boxes bellow if they are true or do not apply:_

- [x] I have performed a self review of my code
- [x] My code follows [Google Java Code style](https://google.github.io/styleguide/javaguide.html)
- [x] I have made my code as simple as possible
- [x] I have added unit tests and the code coverage has not decreased
- [x] I have updated the documentation in all relevant places

### CODE REVIEW

File: src/test/java/eu/bbmri/eric/csit/service/negotiator/unit/service/PostServiceTest.java [hunk 1]
```diff
@@ -29,9 +33,38 @@ public class PostServiceTest {
...
+    publicPost2.setNegotiation(n);
+    publicPost2.setText("public post 2");
+    privatePost1 = new Post();
+    privatePost1.setType(PostType.PRIVATE);
+    privatePost1.setResource(r1);
+    privatePost1.setText("private post 1");
+    privatePost2 = new Post();
+    privatePost2.setType(PostType.PRIVATE);
+    privatePost2.setResource(r2);
+    privatePost2.setText("private post 2");
```
Comment (RadovanTomik) [inline src/test/java/eu/bbmri/eric/csit/service/negotiator/unit/service/PostServiceTest.java:67]: Please refactor this using a builder and a for loop.

**Review by RadovanTomik — CHANGES_REQUESTED** [review]

File: src/main/java/eu/bbmri/eric/csit/service/negotiator/database/model/Post.java [hunk 2]
```diff
@@ -46,5 +46,8 @@ public class Post extends AuditEntity {
   private Resource resource;
 
   @Lob private String text;
+
   private String status;
```
Comment (RadovanTomik) [inline src/main/java/eu/bbmri/eric/csit/service/negotiator/database/model/Post.java:50]: If you have type as Enum then maybe we can do the same for Status?

Comment (asulis) [reply src/main/java/eu/bbmri/eric/csit/service/negotiator/database/model/Post.java:50]: Yes, That's a good idea. I'll fix it

File: src/test/java/eu/bbmri/eric/csit/service/negotiator/unit/service/PostServiceTest.java (hunk 1 above)
Comment (asulis) [reply src/test/java/eu/bbmri/eric/csit/service/negotiator/unit/service/PostServiceTest.java]: Fixed in 5cb22c0

File: src/main/java/eu/bbmri/eric/csit/service/negotiator/database/model/Post.java (hunk 2 above)
Comment (asulis) [reply src/main/java/eu/bbmri/eric/csit/service/negotiator/database/model/Post.java]: Fixed in 5cb22c0

File: src/main/java/eu/bbmri/eric/csit/service/negotiator/database/model/Post.java
Comment (RadovanTomik) [inline src/main/java/eu/bbmri/eric/csit/service/negotiator/database/model/Post.java:1]: Please remove the extension of Audit entity as we are unnecessarily duplicating creator id 3 times.

File: src/main/java/eu/bbmri/eric/csit/service/negotiator/database/model/Post.java [hunk 3]
```diff
@@ -46,5 +46,8 @@ public class Post extends AuditEntity {
   private Resource resource;
 
   @Lob private String text;
-  private String status;
+
+  private PostStatus status;
```
Comment (RadovanTomik) [inline src/main/java/eu/bbmri/eric/csit/service/negotiator/database/model/Post.java:50]: PLease change the persistence of Enums to enum not int. In postgres it is saved as an int.

File: src/test/java/eu/bbmri/eric/csit/service/negotiator/unit/service/PostServiceTest.java [hunk 4]
```diff
@@ -29,9 +35,24 @@ public class PostServiceTest {
...
+  private Post publicPost1;
+  private Post publicPost2;
+  private Post privatePost1;
+  private Post privatePost2;
+
   @BeforeEach
-  void before() {
+  void before() throws IOException {
     closeable = MockitoAnnotations.openMocks(this);
+    Negotiation n = new Negotiation();
```
Comment (RadovanTomik) [inline src/test/java/eu/bbmri/eric/csit/service/negotiator/unit/service/PostServiceTest.java:46]: Please use better variable names

**Review by RadovanTomik — CHANGES_REQUESTED** [review]
Please remove the old private posts entity. After that we can merge.

**Review by RadovanTomik — APPROVED** [review]

**Pull request merged**